flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -192,8 +192,6 @@ public void testWriteBufferToAllSubpartitionsReferenceCounting() throws Exceptio
 
 		partition.writeBufferToAllSubpartitions(buffer);
 
-		// Verify added to all queues, i.e. two buffers in total
-		assertEquals(2, partition.getTotalNumberOfBuffers());
```

Comment:
How about checking this with `ResultPartition#getAllPartitions()` for each subpartition? Otherwise this test doesn't really test anything.

Reply:
Actually it is testing exactly what the test's name suggests. It sets up partition with two subpartitions and asserts `assertTrue(buffer.isRecycled());`  after `writeBufferToAllSubpartitions(buffer)`. 

`assertEquals(2, partition.getTotalNumberOfBuffers());` 

was:

1. first of all redundant (hence it would be copied/paste code), since the fact whether buffer was written to both subpartitions or not is surely tested somewhere else in some broadcast emit unit test.
2. not checking exactly what you intended, since `writeBufferToAllSubpartitions(...)` implemented in one of many buggy ways would still pass this test. To actually test what thick check was intended to do, we should verify that the `buffer` passed to `writeBufferToAllSubpartitions(buffer)` was actually *written* to both partitions - not that statistics were accounted for it, but again, see point 1.